paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/MigrateDatabaseProcedure.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.spark.sql.types.DataTypes.IntegerType;
+import static org.apache.spark.sql.types.DataTypes.StringType;
+
+/**
+ * Migrate database procedure. Usage:
+ *
+ * <pre><code>
+ *  CALL sys.migrate_database(source_type => 'hive', database => 'db01', options => 'x1=y1,x2=y2')
+ * </code></pre>
+ */
+public class MigrateDatabaseProcedure extends BaseProcedure {
+
+    private static final ProcedureParameter[] PARAMETERS =
+            new ProcedureParameter[] {
+                ProcedureParameter.required("source_type", StringType),
+                ProcedureParameter.required("database", StringType),
+                ProcedureParameter.optional("options", StringType),
+                ProcedureParameter.optional(
+                        "options_map", DataTypes.createMapType(StringType, StringType)),
+                ProcedureParameter.optional("parallelism", IntegerType),
+            };
+
+    private static final StructType OUTPUT_TYPE =
+            new StructType(
+                    new StructField[] {
+                        new StructField("result", DataTypes.BooleanType, true, Metadata.empty())
+                    });
+
+    protected MigrateDatabaseProcedure(TableCatalog tableCatalog) {
+        super(tableCatalog);
+    }
+
+    @Override
+    public ProcedureParameter[] parameters() {
+        return PARAMETERS;
+    }
+
+    @Override
+    public StructType outputType() {
+        return OUTPUT_TYPE;
+    }
+
+    @Override
+    public InternalRow[] call(InternalRow args) {
+        String format = args.getString(0);
+        String database = args.getString(1);
+        String properties = args.isNullAt(2) ? null : args.getString(2);
+        MapData mapData = args.isNullAt(3) ? null : args.getMap(3);
+        Map<String, String> optionMap = mapDataToHashMap(mapData);
+        int parallelism =
+                args.isNullAt(4) ? Runtime.getRuntime().availableProcessors() : args.getInt(4);
+
+        Catalog paimonCatalog = ((WithPaimonCatalog) tableCatalog()).paimonCatalog();
+
+        Map<String, String> options = ParameterUtils.parseCommaSeparatedKeyValues(properties);
+        options.putAll(optionMap);
+
+        try {
+            List<Migrator> migrators =
+                    TableMigrationUtils.getImporters(
+                            format, paimonCatalog, database, parallelism, options);
+
+            for (Migrator migrator : migrators) {
+                migrator.executeMigrate();
+                migrator.renameTable(false);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Call migrate_database error: " + e.getMessage(), e);
+        }
```

Comment:
Is it possible to do a catch for each table separately?

Reply:
Good reminder, here dont need try catch, because executeMigrate had handled for every table. @LinMingQiang